unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/WorldChatWindowHUD/WorldChatWindowComponentView.cs

```diff
@@ -0,0 +1,158 @@
+    [SerializeField] private CollapsablePublicChannelListComponentView publicChannelList;
+    [SerializeField] private CollapsableDirectChatListComponentView directChatList;
+    [SerializeField] private CollapsableChatSearchListComponentView searchResultsList;
+    [SerializeField] private Button closeButton;
+    [SerializeField] private GameObject directChatsLoadingContainer;
+    [SerializeField] private GameObject directChatsContainer;
+    [SerializeField] private GameObject directChannelHeader;
+    [SerializeField] private GameObject searchResultsHeader;
+    [SerializeField] private TMP_Text directChatsHeaderLabel;
+    [SerializeField] private TMP_Text searchResultsHeaderLabel;
+    [SerializeField] private ScrollRect scroll;
+    [SerializeField] private SearchBarComponentView searchBar;
+    [SerializeField] private Model model;
+    
+    private string lastSearch;
+
+    public event Action OnClose;
+    public event Action<string> OnOpenPrivateChat;
+    public event Action<string> OnOpenPublicChannel;
+    public event Action<string> OnUnfriend
+    {
+        add => directChatList.OnUnfriend += value;
+        remove => directChatList.OnUnfriend -= value;
+    }
+
+    public RectTransform Transform => (RectTransform) transform;
+    public bool IsActive => gameObject.activeInHierarchy;
+    
+    public static WorldChatWindowComponentView Create()
+    {
+        return Instantiate(Resources.Load<WorldChatWindowComponentView>("SocialBarV1/ConversationListHUD"));
+    }
+
+    public override void Awake()
+    {
+        base.Awake();
+        closeButton.onClick.AddListener(() => OnClose?.Invoke());
+        directChatList.SortingMethod = (a, b) => b.Model.lastMessageTimestamp.CompareTo(a.Model.lastMessageTimestamp);
+        directChatList.OnOpenChat += entry => OnOpenPrivateChat?.Invoke(entry.Model.userId);
+        publicChannelList.OnOpenChat += entry => OnOpenPublicChannel?.Invoke(entry.Model.channelId);
+        searchBar.OnSearchText += Filter;
+        UpdateHeaders();
+    }
+
+    public void Initialize(IChatController chatController, ILastReadMessagesService lastReadMessagesService)
+    {
+        directChatList.Initialize(chatController, lastReadMessagesService);
+        publicChannelList.Initialize(chatController, lastReadMessagesService);
+        searchResultsList.Initialize(chatController, lastReadMessagesService);
+    }
+
+    public void Show() => gameObject.SetActive(true);
+
+    public void Hide() => gameObject.SetActive(false);
+
+    public void SetPrivateChat(PrivateChatModel model)
+    {
+        var user = model.user;
+        directChatList.Set(user.userId, new PrivateChatEntry.PrivateChatEntryModel(
+            user.userId,
+            user.userName,
+            model.recentMessage.body,
+            user.face256SnapshotURL,
+            model.isBlocked,
+            model.isOnline,
+            model.recentMessage.timestamp));
+        directChatList.Sort();
+        UpdateHeaders();
+    }
```

Comment:
You can implement the interface IComponentModelConfig and replace this function by the Configure(BaseComponentModel newModel) one provided.

Reply:
done!